Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Internal/FbOptions.cs

```diff
@@ -33,6 +45,6 @@ public void Validate(IDbContextOptions options)
 		}
 
 		static FbOptionsExtension GetOptions(IDbContextOptions options)
-			=> options.FindExtension<FbOptionsExtension>() ?? new FbOptionsExtension();
+			=> options.Extensions.OfType<FbOptionsExtension>().FirstOrDefault();
```

Comment:
Why?

Reply:
I do not need the new!
You can not find "FindExtension"

And the code gets a lot better!

Reply:
What?

Reply:
I would have to see this in Microsoft.EntityFrameworkCore because it does not return.

Some others use it like this (Npgsql, Mysql (Oracle) ...

And I have noticed that it is indeed the best form!

I tested it and it just does not work in its way.

Reply:
I will test again. I do not know why? More with me I tested a lot before applying this!

Reply:
**He calls here in every way!**

https://github.com/aspnet/EntityFrameworkCore/blob/dev/src/EFCore.SqlServer/Update/Internal/SqlServerModificationCommandBatchFactory.cs#L55

Reply:
This is redundant.
The best way is what I suggested!
is my humble suggestion

Reply:
No it's not the best way. The `FindExtension` is [using hastable lookup internally](https://github.com/aspnet/EntityFrameworkCore/blob/dev/src/EFCore/DbContextOptions.cs#L48), hence it's more efficient. You're iterating through the enumerator.

I suppose the `SqlServerModificationCommandBatchFactory.cs` wasn't updated yet to use the method available.

Reply:
I figured out why it does not work!
I needed a CAST in the "FbDbContextOptionsExtensions.cs" file

**From:** var extension = GetOrCreateExtension (optionsBuilder) .WithConnectionString (connectionString)

**To:** var extension = (FbOptionsExtension) GetOrCreateExtension (optionsBuilder) .WithConnectionString (connectionString);

Reply:
Everything worked! :) rss